dml_events.go

```diff
@@ -406,17 +412,37 @@ func Int64Value(value interface{}) (int64, bool) {
 //
 // ref: https://github.com/mysql/mysql-server/blob/mysql-5.7.5/mysys/charset.c#L963-L1038
 // ref: https://github.com/go-sql-driver/mysql/blob/9181e3a86a19bacd63e68d43ae8b7b36320d8092/utils.go#L717-L758
-func appendEscapedString(buffer []byte, value string) []byte {
+//
+// We need to support right-padding of the generated string using 0-bytes to
+// mimic what a MySQL server would do for BINARY columns (with fixed length).
+//
+// ref: https://github.com/Shopify/ghostferry/pull/159
+//
+// This is specifically mentioned in the the below link:
+//
+//    When BINARY values are stored, they are right-padded with the pad value
+//    to the specified length. The pad value is 0x00 (the zero byte). Values
+//    are right-padded with 0x00 for inserts, and no trailing bytes are removed
+//    for retrievals.
+//
+// ref: https://dev.mysql.com/doc/refman/5.7/en/binary-varbinary.html
+func appendEscapedString(buffer []byte, value string, rightPadLengthForBinaryColumn uint) []byte {
 	buffer = append(buffer, '\'')
 
-	for i := 0; i < len(value); i++ {
+	var i int
+	for i = 0; i < len(value); i++ {
 		c := value[i]
 		if c == '\'' {
 			buffer = append(buffer, '\'', '\'')
 		} else {
 			buffer = append(buffer, c)
 		}
 	}
+	// continue 0-padding up to the desired length as provided by the
+	// caller
+	for ; i < int(rightPadLengthForBinaryColumn); i++ {
+		buffer = append(buffer, '\x00')
```

Comment:
I meant to say that we're supposed to generate MySQL queries here. Is having a `\0`valid in a query? Or should it be `"\0"`? The difference is that the latter is the literal ascii character `'\'` and `'0'` while the former is actually just `0`, if that makes sense.

Reply:
heh, I agree :-) My original code was injecting literal `0` (that is, the byte, not the character), and then I switched it to be `\x00` because it was suggested as improvement.

Still, please note that in neither case are we inserting a 2-character string of `\0` (that is, a `\` and a `0`).

The only difference I can see is that one is an int and the other an int32. **However:** compared to you guys, I'm probably a golang noob and may simply be wrong here.

But, yes, we are building SQL statements, and we want to have literal 0-bytes (not `0` strings or `\0` strings) be injected.